pkg/splunk/enterprise/util.go

```diff
@@ -34,20 +37,248 @@ import (
+// GetVersionedSecretName returns a versioned secret name
+func GetVersionedSecretName(versionedSecretIdentifier string, version string) string {
+	return fmt.Sprintf("%s-secret-v%s", versionedSecretIdentifier, version)
+}
+
+// GetNamespaceScopedSecretName gets namespace scoped secret name
+func GetNamespaceScopedSecretName(namespace string) string {
+	return fmt.Sprintf(namespaceScopedSecretName, namespace)
+}
+
+// GetNamespaceScopedSecret retreives namespace scoped secret
+func GetNamespaceScopedSecret(c splcommon.ControllerClient, namespace string) (*corev1.Secret, error) {
+	var namespaceScopedSecret corev1.Secret
+
+	// Check if a namespace scoped secret exists
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: GetNamespaceScopedSecretName(namespace)}
+	err := c.Get(context.TODO(), namespacedName, &namespaceScopedSecret)
+	if err != nil {
+		// Didn't find it
+		return nil, err
+	}
+
+	return &namespaceScopedSecret, nil
+}
+
+// GetVersionedSecretVersion checks if the secretName includes the versionedSecretIdentifier and if so, extracts the version
+func GetVersionedSecretVersion(secretName string, versionedSecretIdentifier string) (int, error) {
+	// Extracting version from secret's name
+	version := strings.TrimPrefix(secretName, GetVersionedSecretName(versionedSecretIdentifier, ""))
+
+	// Check if the secretName includes the versionedSecretIdentifier
+	if version != secretName {
+		// Includes, version extracted, check if version number is valid
+		versionInt, err := strconv.Atoi(version)
+		if err != nil {
+			return -1, errors.New(nonIntegerVersionError)
+		}
+
+		// Versions should be > 0
+		if versionInt <= 0 {
+			return -1, errors.New(lessThanOrEqualToZeroVersionError)
+		}
+
+		return versionInt, nil
+	}
+
+	// Secret name not matching required criteria
+	return -1, fmt.Errorf(nonMatchingStringError, secretName, versionedSecretIdentifier)
+}
+
+// GetExistingLatestVersionedSecret retreives latest EXISTING versionedSecretIdentifier based secret existing currently in the namespace
+func GetExistingLatestVersionedSecret(c splcommon.ControllerClient, namespace string, versionedSecretIdentifier string) (*corev1.Secret, int, bool) {
+	scopedLog := log.WithName("GetExistingLatestVersionedSecret").WithValues(
+		"versionedSecretIdentifier", versionedSecretIdentifier,
+		"namespace", namespace)
+
+	// Get list of secrets in K8S cluster
+	secretList := corev1.SecretList{}
+
+	// Retreive only secrets only from namespace
+	listOpts := []client.ListOption{
+		client.InNamespace(namespace),
+	}
+
+	err := c.List(context.TODO(), &secretList, listOpts...)
+	if err != nil || len(secretList.Items) == 0 {
+		scopedLog.Error(err, "Secrets not found in namespace")
+		return nil, -1, false
+	}
+
+	// secretFound is true when we find atleast one versionedSecretIdentifier based secret AND we can extract its version successfully
+	var secretFound bool = false
```

Comment:
is this necessary? could we just use `existingLatestVersion != -1` ?

Reply:
and subsequently (for return values) `existingLatestVersionedSecret != null`

Reply:
Done. Removed the bool.